File: client/src/services/employeeService.ts
import { apiClient } from "./apiClient"; // CHANGED: was direct `axios` import + BASE_URL

export type DropdownItem = { id: number; name: string };

export type EmployeeFormOptions = {
    genders: DropdownItem[];
    communities: DropdownItem[];
    designations: DropdownItem[];
    departments: DropdownItem[];
    locations: DropdownItem[];
    shifts: DropdownItem[];
    roles: DropdownItem[];
};

export type Employee = {
    employeeId: number;
    employeeCode: string;
    employeeName: string;
    genderId: number;
    genderName: string;
    communityId: number | null;
    communityName: string | null;
    designationId: number;
    designationName: string;
    departmentId: number;
    departmentName: string;
    locationId: number;
    locationName: string;
    shiftId: number;
    shiftName: string;
    mobileNo: string | null;
    email: string | null;
    address: string | null;
    joiningDate: string | null;
    isActive: boolean;
    username: string;
    roleName: string;
};

export type EmployeeCreatePayload = {
    employeeCode: string;
    employeeName: string;
    genderId: number;
    communityId: number | null;
    designationId: number;
    departmentId: number;
    locationId: number;
    shiftId: number;
    mobileNo: string;
    email: string;
    address: string;
    joiningDate: string | null;
    username: string;
    password: string;
    roleId: number;
};

// Code, Name, Gender are locked — cannot be updated after creation
export type EmployeeUpdatePayload = {
    communityId: number | null;
    designationId: number;
    departmentId: number;
    locationId: number;
    shiftId: number;
    mobileNo: string;
    email: string;
    address: string;
    joiningDate: string | null;
    isActive: boolean;
};

export const getFormOptions = async (): Promise<EmployeeFormOptions> => {
    const response = await apiClient.get(`/employee/form-options`);
    return response.data;
};

export const getEmployees = async (username: string): Promise<Employee[]> => {
    const response = await apiClient.get(`/employee`, {
        params: { username }
    });
    return response.data;
};

export const searchEmployees = async (keyword: string, username: string): Promise<Employee[]> => {
    const response = await apiClient.get(`/employee/search`, {
        params: { keyword, username }
    });
    return response.data;
};

export const createEmployee = async (payload: EmployeeCreatePayload) => {
    console.log("CREATE PAYLOAD:", JSON.stringify(payload));
    const response = await apiClient.post(`/employee`, payload);
    return response.data;
};

export const updateEmployee = async (
    employeeId: number,
    payload: EmployeeUpdatePayload
) => {
    const response = await apiClient.put(
        `/employee/${employeeId}`,
        payload
    );
    return response.data;
};

File: AttendanceAPI/Services/EmployeeService.cs
using AttendanceAPI.DTOs;
using AttendanceAPI.Repositories.Interfaces;
using AttendanceAPI.Services.Interfaces;
using AttendanceAPI.Models;
using Microsoft.AspNetCore.Identity;

namespace AttendanceAPI.Services;

public class EmployeeService : IEmployeeService
{
    private readonly IEmployeeRepository _repository;
    private readonly IAuditLogService _auditLogService;
    private readonly DataProtectionService _dataProtection;

    public EmployeeService(
        IEmployeeRepository repository,
        IAuditLogService auditLogService,
        DataProtectionService dataProtection)
    {
        _repository = repository;
        _auditLogService = auditLogService;
        _dataProtection = dataProtection;
    }

    public async Task<EmployeeFormOptionsDto> GetFormOptionsAsync()
    {
        return await _repository.GetFormOptionsAsync();
    }

    public async Task<bool> UpdateEmployeeAsync(int employeeId, EmployeeUpdateRequestDto request)
    {
        var employee = await _repository.GetEmployeeByIdAsync(employeeId);
        if (employee == null) return false;

        employee.CommunityId   = request.CommunityId;
        employee.DesignationId = request.DesignationId;
        employee.DepartmentId  = request.DepartmentId;
        employee.LocationId    = request.LocationId;
        employee.ShiftId       = request.ShiftId;
        employee.MobileNo      = string.IsNullOrWhiteSpace(request.MobileNo) ? null : _dataProtection.Encrypt(request.MobileNo);
        employee.Email         = string.IsNullOrWhiteSpace(request.Email) ? null : _dataProtection.Encrypt(request.Email);
        employee.Address       = string.IsNullOrWhiteSpace(request.Address) ? null : _dataProtection.Encrypt(request.Address);
        employee.JoiningDate   = request.JoiningDate;
        employee.IsActive      = request.IsActive;

        await _repository.SaveAsync();
        return true;
    }

    public async Task<List<EmployeeResponseDto>> GetActiveEmployeesAsync(string username)
    {
        var roleId = await _repository.GetRoleIdByUsernameAsync(username);
        var employees = await _repository.GetActiveEmployeesAsync(roleId ?? 4);
        return employees.Select(MapToDto).ToList();
    }

    public async Task<List<EmployeeResponseDto>> GetEmployeesByCodeAsync(string employeeCode)
    {
        var employees = await _repository.GetEmployeesByCodeAsync(employeeCode);
        return employees.Select(MapToDto).ToList();
    }

    public async Task<(bool Success, string Message, EmployeeResponseDto? Employee)>
        CreateEmployeeAsync(EmployeeCreateRequestDto request)
    {
        if (await _repository.EmployeeCodeExistsAsync(request.EmployeeCode))
            return (false, $"Employee code '{request.EmployeeCode}' already exists", null);

        if (await _repository.UsernameExistsAsync(request.Username))
            return (false, $"Username '{request.Username}' already exists", null);

        var employee = new EmployeeMaster
        {
            EmployeeCode  = request.EmployeeCode,
            EmployeeName  = request.EmployeeName,
            GenderId      = request.GenderId,
            CommunityId   = request.CommunityId,
            DesignationId = request.DesignationId,
            DepartmentId  = request.DepartmentId,
            LocationId    = request.LocationId,
            ShiftId       = request.ShiftId,
            MobileNo      = string.IsNullOrWhiteSpace(request.MobileNo) ? null : _dataProtection.Encrypt(request.MobileNo),
            Email         = string.IsNullOrWhiteSpace(request.Email) ? null : _dataProtection.Encrypt(request.Email),
            Address       = string.IsNullOrWhiteSpace(request.Address) ? null : _dataProtection.Encrypt(request.Address),
            JoiningDate   = request.JoiningDate,
            IsActive      = true,
            CreatedAt     = DateTime.Now,
            UpdatedAt     = DateTime.Now
        };

        var created = await _repository.CreateEmployeeAsync(employee);

        var passwordHasher = new PasswordHasher<UserLogin>();
        var login = new UserLogin
        {
            EmployeeId = created.EmployeeId,
            Username   = request.Username,
            RoleId     = request.RoleId,
            IsActive   = true
        };
        login.PasswordHash = passwordHasher.HashPassword(login, request.Password);

        await _repository.CreateUserLoginAsync(login);

        await _auditLogService.LogAsync(
            login.UserId,
            "INSERT",
            "employee_master",
            created.EmployeeId,
            $"Employee {created.EmployeeCode} created"
        );

        var dto = MapToDto(created);
        dto.Username = login.Username;
        return (true, "Employee created successfully", dto);
    }

    public async Task<List<EmployeeResponseDto>> SearchEmployeesAsync(string keyword)
    {
        var employees = await _repository.SearchEmployeesAsync(keyword);
        return employees.Select(MapToDto).ToList();
    }

    private EmployeeResponseDto MapToDto(EmployeeMaster e) => new()
    {
        EmployeeId      = e.EmployeeId,
        EmployeeCode    = e.EmployeeCode,
        EmployeeName    = e.EmployeeName,
        GenderName      = e.Gender?.GenderName,
        CommunityName   = e.Community?.CommunityName,
        DesignationName = e.Designation?.DesignationName,
        DepartmentName  = e.Department?.DepartmentName,
        LocationName    = e.Location?.LocationName,
        ShiftName       = e.Shift?.ShiftName,
        MobileNo        = string.IsNullOrEmpty(e.MobileNo) ? null : _dataProtection.Decrypt(e.MobileNo),
        Email           = string.IsNullOrEmpty(e.Email) ? null : _dataProtection.Decrypt(e.Email),
        Address         = string.IsNullOrEmpty(e.Address) ? null : _dataProtection.Decrypt(e.Address),
        JoiningDate     = e.JoiningDate,
        IsActive        = e.IsActive,
        Username        = e.Username,
        RoleName        = e.RoleName
    };
}